core/finality-grandpa/primitives/src/lib.rs

```diff
@@ -21,17 +21,31 @@
 #[cfg(not(feature = "std"))]
 extern crate alloc;
 
+#[cfg(feature = "std")]
+use serde::Serialize;
 use parity_codec::{Encode, Decode};
-use substrate_primitives::ed25519;
-use sr_primitives::traits::{DigestFor, NumberFor};
+use sr_primitives::{ConsensusEngineId, traits::{DigestFor, NumberFor}};
 use client::decl_runtime_apis;
 use rstd::vec::Vec;
 
-use ed25519::Public as AuthorityId;
+/// The grandpa crypto scheme defined via the keypair type.
+pub type AuthorityPair = substrate_primitives::ed25519::Pair;
+
+/// Identity of a Grandpa authority.
+pub type AuthorityId = substrate_primitives::ed25519::Public;
+
+/// Signature for a Grandpa authority.
+pub type AuthoritySignature = substrate_primitives::ed25519::Signature;
+
+/// The `ConsensusEngineId` of BABE.
+pub const GRANDPA_ENGINE_ID: ConsensusEngineId = *b"FRNK";
```

Comment:
BTW, there is already one defined elsewhere (and not as FRNK).

```
core/finality-grandpa/src/communication/mod.rs:pub const GRANDPA_ENGINE_ID: ConsensusEngineId = [b'a', b'f', b'g', b'1'];
```

Reply:
@gavofyork this hasn't been resolved. why did you mark it as such? What is "FRNK"?

Reply:
This has been resolved.